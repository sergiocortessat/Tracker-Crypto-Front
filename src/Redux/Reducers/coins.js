const initialState = {
  coins: [],
};

const coinReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_COINS':
      return {
        ...state,
        coins: action.payload,
      };
    default:
      return state;
  }
};

export default coinReducer;
