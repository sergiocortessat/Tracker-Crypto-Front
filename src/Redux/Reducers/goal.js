const initialState = {
  goals: [],
};

const goalReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_GOALS':
      return {
        ...state,
        goal: action.payload,
      };
    default:
      return state;
  }
};

export default goalReducer;
