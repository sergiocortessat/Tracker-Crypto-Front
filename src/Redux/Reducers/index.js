import { combineReducers } from 'redux';
import Coins from './coins';
import User from './user';

const reducers = combineReducers({
  coins: Coins,
  user: User,
});

export default reducers;
