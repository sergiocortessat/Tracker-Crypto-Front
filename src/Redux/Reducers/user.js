const initialState = {
  user: [],
};

const userReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'CURRENT_USER':
      return {
        ...state,
        user: action.payload,
      };
    default:
      return state;
  }
};

export default userReducer;
