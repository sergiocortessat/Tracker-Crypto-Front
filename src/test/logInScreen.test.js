import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import Reducers from '../Redux/Reducers';
import LogIn from '../components/LogInScreen';

const store = createStore(Reducers);

it('renders correctly', () => {
  const addMeasurement = renderer
    .create(
      <Provider store={store}>
        <LogIn />
      </Provider>,
    )
    .toJSON();
  expect(addMeasurement).toMatchSnapshot();
});
