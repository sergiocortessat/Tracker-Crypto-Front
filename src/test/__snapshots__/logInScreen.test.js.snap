// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="log-in-container"
>
  <h1>
    Please login
  </h1>
  <button
    className="auth-button"
    onClick={[Function]}
    type="button"
  >
    <span>
      Log In
    </span>
  </button>
</div>
`;
